### PR TITLE
Address Feedback fix spring boot TODOs

### PR DESCRIPTION
**Pull request by frankgh**
- install-dev -> install-server
- fix expected output on some files
- refactor trace/hint information on the c-side
- remove path from Writable/stream endpoint as it is no longer used
- refactor PxfErrorAttributes
- address spring_boot_todo
- expose only a few endpoints (health, info, shutdown)

### CODE REVIEW

File: external-table/src/libchurl.c [hunk 1]
```diff
@@ -640,12 +640,12 @@ flush_internal_buffer(churl_context *context)
 		multi_perform(context);
 	}
 
+	check_response(context);
+
```
Comment (denalex) [inline external-table/src/libchurl.c:644]: wonder why this was needed

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java [hunk 2]
```diff
@@ -9,30 +10,64 @@
...
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
 /**
  * Declares the registerSecurityServletFilter bean method to be processed by
  * the Spring container
  */
 @Configuration
 @EnableConfigurationProperties(PxfServerProperties.class)
-public class PxfConfiguration {
+public class PxfConfiguration implements WebMvcConfigurer {
```
Comment (denalex) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java:27]: the comment above lines 21-24 is no longer relevant (registerSecurityServletFilter)

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java [hunk 3]
```diff
@@ -9,30 +10,64 @@
...
+     * Configures and builds the {@link ThreadPoolTaskExecutor}
      *
-     * @return the {@link TaskExecutorBuilder} object
+     * @return the {@link ThreadPoolTaskExecutor}
      */
-    @Bean
-    public TaskExecutorBuilder taskExecutorBuilder(PxfServerProperties pxfServerProperties,
-                                                   ObjectProvider<TaskExecutorCustomizer> taskExecutorCustomizers,
-                                                   ObjectProvider<TaskDecorator> taskDecorator) {
+    @Lazy
```
Comment (denalex) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java:64]: I thought we did not want lazy init unless absolutely required

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java [hunk 4]
```diff
@@ -9,30 +10,64 @@
...
  */
 @Configuration
 @EnableConfigurationProperties(PxfServerProperties.class)
-public class PxfConfiguration {
+public class PxfConfiguration implements WebMvcConfigurer {
+
+    /**
+     * Bean name of PXF's {@link TaskExecutor}.
+     */
+    public static final String PXF_APPLICATION_TASK_EXECUTOR_BEAN_NAME = "pxfApplicationTaskExecutor";
```
Comment (denalex) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java:32]: can we fine-tune a name -- it is not executing any arbitrary pxf tasks, this is for async response streaming only, in other words we should not be submitting any arbitrary async work here, right ? Maybe `pxfResponseStreamTaskExecutor` or smith even simpler

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java [hunk 5]
```diff
@@ -9,30 +10,64 @@
...
     /**
-     * Configures the TaskExecutorBuilder to be used for async requests (i.e. Bridge
+     * Configures the TaskExecutor to be used for async requests (i.e. Bridge
      * Read).
+     */
+    @Override
+    public void configureAsyncSupport(AsyncSupportConfigurer configurer) {
+        Object taskExecutor = this.beanFactory
+                .getBean(PXF_APPLICATION_TASK_EXECUTOR_BEAN_NAME);
+        if (taskExecutor instanceof AsyncTaskExecutor) {
```
Comment (denalex) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java:54]: why `if` ? We expect our own bean here, so cast or let it fail the startup if somehow misconfigured

File: external-table/src/libchurl.c (hunk 1 above)
Comment (frankgh) [reply external-table/src/libchurl.c:644]: in one case the response was closed by the server, but we already had a response in the buffer, so we could actually take some action in that case. For example, check the response code

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java (hunk 3 above)
Comment (frankgh) [reply server/pxf-service/src/main/java/org/greenplum/pxf/service/PxfConfiguration.java:64]: Yeah, I'll address it

**Comment by frankgh** [comment]
@denalex thanks for the review. I have addressed your concerns

**Pull request merged**